ystia/yorc
# Fix wrong state on purge error

# Pull Request description

## Description of the change

Detect errors on purge an set deployment state to `UNDEPLOYMENT_FAILED` and task state to `FAILED`

### How to verify it

1. Deploy a very basic app
2. Change ownership of the work/deployments/<depID> folder to something else than Yorc's ownership (root will do)
3. Purge the deployment

Deployment status should be set  to `UNDEPLOYMENT_FAILED` and task state to `FAILED`. Restoring ownership and running the purge again should work.

### Description for the changelog

* An error during deployment purge may let the deployment in a wrong state ([GH-572](https://github.com/ystia/yorc/issues/572))

## Applicable Issues

Fixes #572
